pkg/volume/volume.go

```diff
@@ -207,6 +248,108 @@ func (v *Operation) Create() (*v1alpha1.CASVolume, error) {
 	return vol, nil
 }
 
+// Resize updates the size of a CASVolume(cStor)
+func (v *Operation) Resize() (*v1alpha1.CASVolume, error) {
+	glog.V(4).Infof("fetching the details of pvc,pv,sc,casName")
+
+	capacityValue, capacityUnit, err := splitString(v.volume.Spec.Capacity)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to process Capacity")
+	}
+	newSize, err := RoundUpStringToBytes(capacityValue, capacityUnit)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to convert capacity '%s' to Bytes", v.volume.Spec.Capacity)
+	}
+
+	// pv details
+	pv, err := v.k8sClient.GetPV(v.volume.Name, mach_apis_meta_v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	oldSize := RoundUpToBytes(pv.Spec.Capacity[core_v1.ResourceStorage])
+	glog.V(4).Infof("old capacity: '%d' and requested capacity: '%d' in Bytes", oldSize, newSize)
+
+	// If volume size is already greater than or equal to requested size return
+	if uint64(oldSize) >= newSize {
+		return nil, fmt.Errorf("Volume is already greater than or equal to requested volume size")
+	}
+
+	if string(v.volume.Namespace) != string(pv.Spec.ClaimRef.Namespace) {
```

Comment:
Namespace should be fetched and updated from the ClaimRef. CSI will not be able to pass the namespace for resize. Only PV name will be received.